src/class-scripts.php

```diff
@@ -0,0 +1,205 @@
+		if ( $this->parsely->api_key_is_missing() || $parsely_options['disable_javascript'] ) {
+			return;
+		}
+
+		global $post;
+		$display = true;
+		if ( in_array( get_post_type(), $parsely_options['track_post_types'], true ) && ! Parsely::post_has_trackable_status( $post ) ) {
+			$display = false;
+		}
+		if ( ! $parsely_options['track_authenticated_users'] && $this->parsely->parsely_is_user_logged_in() ) {
+			$display = false;
+		}
+		if ( ! in_array( get_post_type(), $parsely_options['track_post_types'], true ) && ! in_array( get_post_type(), $parsely_options['track_page_types'], true ) ) {
+			$display = false;
+		}
+
+		/**
+		 * Filters whether to enqueue the Parsely JavaScript tracking script from the CDN.
+		 *
+		 * If true, the script is enqueued.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param bool $display True if the JavaScript file should be included. False if not.
+		 */
+		if ( ! apply_filters( 'wp_parsely_load_js_tracker', $display ) ) {
+			return;
+		}
+
+		if ( ! has_filter( 'script_loader_tag', array( $this, 'script_loader_tag' ) ) ) {
+			add_filter( 'script_loader_tag', array( $this, 'script_loader_tag' ), 10, 3 );
+		}
+
+		wp_enqueue_script( 'wp-parsely-tracker' );
+	}
+
+	/**
+	 * Load JavaScript for Parse.ly API.
+	 *
+	 * @since 2.5.0
+	 *
+	 * @return void
+	 */
+	public function load_js_api(): void {
+		$parsely_options = $this->parsely->get_options();
+
+		// If we don't have an API secret, there's no need to proceed.
+		if ( empty( $parsely_options['api_secret'] ) ) {
+			return;
+		}
+
+		if ( ! has_filter( 'script_loader_tag', array( $this, 'script_loader_tag' ) ) ) {
+			add_filter( 'script_loader_tag', array( $this, 'script_loader_tag' ), 10, 3 );
+		}
+
+		wp_enqueue_script( 'wp-parsely-api' );
+	}
+
+	/**
+	 * Filter the script tag for certain scripts to add needed attributes.
+	 *
+	 * @since 2.5.0
+	 *
+	 * @param string $tag    The `script` tag for the enqueued script.
+	 * @param string $handle The script's registered handle.
+	 * @param string $src    Unused? The script's source URL.
+	 * @return string Amended `script` tag.
+	 */
+	public function script_loader_tag( string $tag, string $handle, string $src ): string {
+		$parsely_options = $this->parsely->get_options();
+		if ( in_array(
+			$handle,
+			array(
+				'wp-parsely',
+				'wp-parsely-api',
+				'wp-parsely-tracker',
+				'wp-parsely-recommended-widget',
+			)
+		) ) {
+			/**
+			 * Filter whether to include the CloudFlare Rocket Loader attribute (`data-cfasync=false`) in the script.
+			 * Only needed if the site being served is behind Cloudflare. Should return false otherwise.
+			 * https://support.cloudflare.com/hc/en-us/articles/200169436-How-can-I-have-Rocket-Loader-ignore-specific-JavaScripts
+			 *
+			 * @since 3.0.0
+			 *
+			 * @param bool $enabled True if enabled, false if not.
+			 * @param string $handle The script's registered handle.
+			 */
+			if ( apply_filters( 'wp_parsely_enable_cfasync_attribute', false, $handle ) ) {
```

Comment:
Not worth blocking on, but worth noting that for a lot of Wordpress users they have no / limited dev resources: it might be worth adding a setting in the UI for this as well as having it in a filter, so non-developer clients have an option for fixing this too.

Reply:
I see your point, but for this specific issue I'd say it's OK to have a filter. Having a UI option requires a database access and more complex logic, for a feature that we'll probably want to have disabled anyway. 

Given that Cloudflare is a developer-centric feature, I'd leave it as is.